### DISCUSSION TITLE
JS events not triggered on Cart page

### DISCUSSION THREAD

**Discussion by ChadReitsma** [woocommerce/woocommerce#51526] [Checkout Flow]
Seems like the legacy events (which were extremely nice) are not triggered with the new block cart.  When the QTY is changed the body events don't fire.  I can see that they're attached, but nothing happens.

I have tried using:
1. removed_from_cart
2. wc-blocks_removed_from_cart

Example jQuery code: 
`$('body').on( 'removed_from_cart wc-blocks_removed_from_cart', function() {
	console.log('Should trigger on update product qty on cart page!');	
});`

**EDIT:** I understand that the new Cart/Checkout blocks use React and the Store API, but I still see the wc-block_ events attached to .body, so why aren't they firing?

Cheers,
C.

**Comment by ChadReitsma** [comment]
We just went back to the classic cart + checkout.  

In my opinion this React stuff looks nice - but you are creating solutions for problems that don't even exist. It is much more difficult to customize now. I understand the allure of using newer technologies, but this seems like overkill. Not sure why you guys chose this path... What problems are you solving?

**Reply by opr** [reply]
@ChadReitsma thanks for your feedback, and patience while waiting for a response.

> In my opinion this React stuff looks nice

Thanks, glad to hear that! We are introducing newer designs focused on conversion and shopper experience, so glad you think they look good :)

> but you are creating solutions for problems that don't even exist.

I don't agree on this point. The Cart/Checkout blocks solve for many problems that exist in the shortcode equivalents, most notably ones introduced by actions/filters. These really limit the changes we can make as we have to be very careful not to break backward compatibility in these areas, therefore, changes/improvements often result in code that's difficult to scale or maintain due to the workarounds we have to put in place not to break existing integrations.

> It is much more difficult to customize now.

Understood, and agreed. This is to some extent intentional, as we are really focusing on improving conversion rate, and believe that allowing changes to certain areas could harm this. With that said, we are working on unlocking more customisation options for merchants. This is still a work in progress.

> I understand the allure of using newer technologies, but this seems like overkill. Not sure why you guys chose this path... What problems are you solving?

Hopefully what I mentioned above gives some insight into what we're solving, without having to dive into specific examples. We are also trying to align more with the direction of WordPress and Gutenberg, so developer experience is similar and you don't have to shift into different "systems" while working on one site.

I see you reverted to the shortcodes, and I'm glad they're working for you but I just wanted to note that:

1. Knowing when a quantity changes, or an item has been removed is possible by observing the `wc/store/cart` data store, which aligns more with the patterns used in WordPress Core and Gutenberg.
2. We did discuss re-implementing the legacy events in the blocks, such as the ones in your post. These events will still use the data store mentioned in point one, but would be more familiar to folks who worked with shortcodes before. If we do implement this change we will post an update on the [Woo developer blog](https://developer.woocommerce.com/blog), so please keep an eye out for that.

**Reply by ChadReitsma** [reply]
@opr - Hey Thomas, appreciate your detailed reply!

I guess my frustration stems (mostly) from the lack of documentation for this stuff.  Where can I find detailed information or code examples on "observing the wc/store/cart data store"?  For me, it's much easier to dive into the /woocommerce/templates/ folder to find what I'm looking for.
